### PR TITLE
fix(menu): update location when dynamically changing side or doc dir

### LINKED ISSUE #25601: bug: dynamically changing side prop for menu not working
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [x] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

When I change side props from start to end, it doesn't change

### Expected Behavior

When I change the side props from start to end, it re-renders at the end

### Steps to Reproduce

change language in the code below

### Code Reproduction URL

https://codesandbox.io/s/frosty-http-nnblwg

### Ionic Info

[WARN] Error loading @capacitor/ios package.json: Error: Cannot find module
       '@capacitor/ios/package'
       
       Require stack:
       - /usr/lib/node_modules/@ionic/cli/lib/project/index.js
       - /usr/lib/node_modules/@ionic/cli/lib/index.js
       - /usr/lib/node_modules/@ionic/cli/index.js
       - /usr/lib/node_modules/@ionic/cli/bin/ionic

Ionic:

   Ionic CLI       : 6.20.1 (/usr/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 6.1.13

Capacitor:

   Capacitor CLI      : 3.6.0
   @capacitor/android : 3.6.0
   @capacitor/core    : 3.6.0
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.6.0

System:

   NodeJS : v16.15.0 (/home/abdulmlik/.nvm/versions/node/v16.15.0/bin/node)
   npm    : 8.13.2
   OS     : Linux 5.15

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by thetaPC**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- When the document direction is dynamically changed, the menu does not switch to the correct position. For example, `<ion-menu side='start'>` will always be on the left side regardless of direction.
- When the side property is dynamically changed, the animation starts in the middle of the screen instead of off screen.

Issue URL: resolves #25601 #19489 

## What is the new behavior?

- Animation is smooth regardless of side property or document direction being dynamically change. 
- Menu appears on the correct side based on document direction when dynamically updated.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

N/A

### CODE REVIEW

**Comment by thetaPC** [comment]
@amandaejohnston I've updated the references and the PR is ready to go.

File: core/src/components/menu/test/basic/menu.e2e.ts [hunk 1]
```diff
@@ -74,6 +74,53 @@ test.describe('menu: basic', () => {
 
     await expect(scrollTop).toBe(200);
   });
+
+  test('should render on the correct side when side is changed dynamically', async ({ page }) => {
```
Comment (amandaejohnston) [inline core/src/components/menu/test/basic/menu.e2e.ts:78]: Could we also add `skip.mode('md')` or `skip.mode('ios')` to both tests? These checks aren't mode-dependent, and the more screenshots we can trim down, the better. You can also just manually delete the screenshots for the mode you skipped.

**Review by amandaejohnston — CHANGES_REQUESTED** [review]
This looks great! Just had a small request on the tests.

**Review by amandaejohnston — APPROVED** [review]
Nice work!

**Pull request merged**